code/modules/surgery/plastic_surgery.dm

```diff
@@ -42,8 +69,14 @@
 	else
 		var/list/names = list()
 		if(!isabductor(user))
-			for(var/i in 1 to 10)
-				names += target.dna.species.random_name(target.gender, TRUE)
+			var/obj/item/offhand = user.get_inactive_held_item()
+			if(istype(offhand, /obj/item/photo))
+				var/obj/item/photo/disguises = offhand
+				for(var/mob/namelist as anything in disguises.picture.real_names_seen)
```

Comment:
```suggestion
				for(var/namelist in disguises.picture?.real_names_seen)
```
IIRC you can have photos with no picture datum, like those found in maint? Might want to be safe about that